mlflow/pyfunc/__init__.py

```diff
@@ -3035,9 +3084,16 @@ def predict(self, context, model_input: List[str], params=None) -> List[str]:
                 extra={"color": "red"},
             )
         mlflow_model.signature = signature_from_type_hints
-    # TODO: if signature is provided, we should validate input_example against it
     elif signature:
         mlflow_model.signature = signature
+        if type_hint_from_example:
+            if saved_example is None:
+                _logger.warning(
+                    "Input example must be provided when using TypeFromExample as type hint."
```

Comment:
Are we expecting the user to do something here to remove this warning in subsequent iterations? If so, let's expand this warning to guide them on the proper use of this special type hint.